include/memory/memory.h

```diff
@@ -1,6 +1,14 @@
 #ifndef MEMORY_H_
 #define MEMORY_H_
 
-int memory_flash_region(const void *vAddress, const void *vData, unsigned int length);
+typedef enum {
```

Comment:
I did the analysis here and there are many areas in the code that use 0==success semantics, so converting to bool is not trivial at this stage of the game for 2.8.0 release. recommend shelving the bool conversion to later.